### ISSUE TITLE
Regular expressions

### ISSUE THREAD

**Issue by Simn** [waneck/haxe-genc#10]
We don't really want a PCRE depdendency...

**Comment by Simn** [comment]
@ousado: If I remember correctly you were planning to use some library which has support for regex?

**Comment by ousado** [comment]
yes, at least for Unicode I think it would be wise to use ICU ( http://site.icu-project.org/ ) "ICU's regular expressions fully support Unicode while providing very competitive performance." Haven't looked at the features/compatibility with engines used by other haxe targets so far.

**Comment by waneck** [comment]
Well, even though ICU seems like a beast to me, I think that's the best way to quickly support best practices for string encoding and regex. We can try to simplify that later on.
There's also that google engine, re2, which is reportedly very fast. It seems pretty lightweight, and even though it's C++, we can make a simple interface to use with C.
